### ISSUE TITLE
Wallpaper rotation in kbfs on MacOS locks up preferences

### ISSUE THREAD

**Issue by sheldon** [keybase/client#8297]
Hi

I moved my [wallpaper collection](https://keybase.pub/sheldonels/Wallpaper/) into kbfs to easily share among computers. It was working ok occasionally, but most of the time on MacOS trying to choose that folder in the "Desktop & Screen Saver" preferences panel as a source for Desktop wallpaper would beachball forever, needing to force close preference eventually.

I've now moved back to sharing via other means, but I thought it best to log the bug as I really like kbfs, and hope it can eventually replace other file sharing solutions for me.

I suspect the issue boils down to slow file listing access or something similar. Even in finder, accessing these files is quite slow. If they're part of my own account, are they stored locally as normal files? If my machine has the key to decrypt them anyway, storing encrypted locally might not make much sense security wise, and performance might be an issue.

Any help debugging would be appreciated.

Cheers
Sheldon

**Comment by strib** [comment]
Hi @sheldon.  Sorry for the bad experience.  File access in KBFS doesn't currently support offline access.  We do our best to cache stuff locally, but still there is some network access required once in a while when you access the files.  So my guess is that the beachball happens when either you're offline or having slow network latency.  It probably doesn't have to do with the actual encryption.

Offline support is on our list of things to do, but I'm not sure when exactly that will happen.

Then again, if it really beachballed "forever", and you weren't offline at the time, then maybe something else is going on.

If you want more help debugging, you can do a `keybase log send`, and then let me know the resulting log ID and the time that one of these beachballs happens.

**Comment by sheldon** [comment]
Hey @strib 

I was online most of the time when trying this. I suspect it's exactly what you're saying, that it might just be waiting on the needed download of each file while scanning the folder, and being over 1000 entries with pretty big files, it just takes a very long time. The right solve is probably offline access, so feel free to mark this a duplicate of that issue and close it out.

I'll +1 on offline access and wait for it to come. Thanks again for the awesome work you and your team are doing, I really hope keybase catches on more and we get better encryption everywhere!

Cheers
Sheldon
